runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/HotColdClue.java

```diff
@@ -0,0 +1,351 @@
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.cluescrolls.clues;
+
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.Graphics2D;
+import java.awt.Point;
+import java.awt.Polygon;
+import java.awt.Rectangle;
+import java.awt.geom.Rectangle2D;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import lombok.Getter;
+import net.runelite.api.NPC;
+import net.runelite.api.Perspective;
+import net.runelite.api.coords.LocalPoint;
+import net.runelite.api.coords.WorldPoint;
+import static net.runelite.client.plugins.cluescrolls.ClueScrollOverlay.TITLED_CONTENT_COLOR;
+import net.runelite.client.plugins.cluescrolls.ClueScrollPlugin;
+import static net.runelite.client.plugins.cluescrolls.ClueScrollPlugin.CLUE_SCROLL_IMAGE;
+import static net.runelite.client.plugins.cluescrolls.ClueScrollPlugin.SPADE_IMAGE;
+import static net.runelite.client.plugins.cluescrolls.ClueScrollWorldOverlay.IMAGE_Z_OFFSET;
+import net.runelite.client.plugins.cluescrolls.clues.hotcold.HotColdArea;
+import net.runelite.client.plugins.cluescrolls.clues.hotcold.HotColdLocation;
+import net.runelite.client.ui.overlay.OverlayUtil;
+import net.runelite.client.ui.overlay.components.LineComponent;
+import net.runelite.client.ui.overlay.components.PanelComponent;
+import net.runelite.client.ui.overlay.components.TitleComponent;
+
+@Getter
+public class HotColdClue extends ClueScroll implements TextClueScroll, NpcClueScroll
+{
+	private static final HotColdClue CLUE =
+		new HotColdClue("Buried beneath the ground, who knows where it's found. Lucky for you, A man called Jorral may have a clue.",
+			"Jorral",
+			"Speak to Jorral to receive a strange device.");
+
+	// list of potential places to dig
+	private List<HotColdLocation> digLocations = new ArrayList<>();
+	private String text;
+	private String npc;
+	private String solution;
+	private WorldPoint location;
+	private WorldPoint lastWorldPoint;
+
+	private HotColdClue(String text, String npc, String solution)
+	{
+		this.text = text;
+		this.npc = npc;
+		this.solution = solution;
+		this.location = null;
+	}
+
+	@Override
+	public void makeOverlayHint(PanelComponent panelComponent, ClueScrollPlugin plugin)
+	{
+		panelComponent.getChildren().add(TitleComponent.builder()
+			.text("Hot/Cold Clue")
+			.build());
+		// strange device has not been tested yet, show how to get it
+		if (lastWorldPoint == null && location == null)
+		{
+			panelComponent.setPreferredSize(new Dimension(200, 0));
```

Comment:
Just set the preferred size before the ifs, you are setting same preferred size in both if cases